### ISSUE TITLE
Unable to set event-scoped custom dimension that applies to all sessions and events

### ISSUE THREAD

**Issue by benfixit** [PriceRunner/react-ga4#46]
There is a similar issue reported [here](https://github.com/PriceRunner/react-ga4/issues/31). However, the suggested solution only addresses user-scoped custom dimensions.

Also, the documentation for `ReactGA.event(options)` suggests that `dimension` and `metric` can be passed in but this is not correct as the expected parameter is of type `UaEventOptions`.

```
    export type UaEventOptions = {
        action: string;
        category: string;
        label?: string;
        value?: number;
        nonInteraction?: boolean;
        transport?: ('beacon' | 'xhr' | 'image');
    };
```

**Comment by cabaret** [comment]
We're also running into this issue. The current types do not allow to pass dimensions and metrics into `.event`.

The types over here do support those parameters. Can they be included in this package's types? 

https://github.com/react-ga/react-ga/blob/master/types/index.d.ts#L40

**Comment by BearCooder** [comment]
Yes same here, will there be a solution to support the parameters? @codler 
This package is currently the only one for GA4

**Comment by marioGMIT** [comment]
any one here know if it is possible to use this package to send user properties?

**Comment by blainegarrett** [comment]
I'm running into the same issue. 
It appears to send the dimension if they are named 'dimension1', 'dimension2', etc. 
I suppose this would make sense for migrating from reactGA as this was the required naming convention in UA.
